mining.go

```diff
@@ -1365,7 +1380,7 @@ mempoolLoop:
 
 		// Skip if we already have too many SStx.
 		if isSStx && (numSStx >=
-			int(server.chainParams.MaxFreshStakePerBlock)) {
+			int(g.blockManager.server.chainParams.MaxFreshStakePerBlock)) {
```

Comment:
Please make the creation of the block template generator takes the chain params and use it directly from `g`.  A large reason for having the block template generator is to reduce the dependencies on the other types so that it can be more easily split into a separate package.